apps/studio/data/database/schemas-query.ts

```diff
@@ -39,3 +44,20 @@ export const useSchemasQuery = <TData = SchemasData>(
 export function invalidateSchemasQuery(client: QueryClient, projectRef: string | undefined) {
   return client.invalidateQueries(sqlKeys.query(projectRef, ['schemas', 'list']))
 }
+
+export function prefetchSchemas(
```

Comment:
this is just a thought - should we follow the same api as what you wrote for `usePrefetchEditorTablePage` ? 

so this becomes a hook that returns a callback, then we don't need to separately declare `const queryClient = useQueryClient()` and pass it in - queryClient can be declared in usePrefetchSchemas instead, one prop less

Reply:
^ will apply for `prefetchEntityType` too